boards/BMS/Test/Src/Test_StateMachine.cpp

```diff
@@ -648,4 +628,117 @@ TEST_F(BmsStateMachineTest, check_remains_in_fault_state_until_fault_cleared_the
     LetTimePass(state_machine, 10);
     ASSERT_EQ(App_GetInitState(), App_SharedStateMachine_GetCurrentState(state_machine));
 }
+
+TEST_F(BmsStateMachineTest, check_precharge_state_transitions_and_air_plus_status)
+{
+    struct
+    {
+        bool  air_negative_closes;
+        float initial_ts_voltage;
+        float precharge_duration;
+        bool  expect_precharge_starts;
+        bool  expect_precharge_successful;
+    } test_params[5] = { {
```

Comment:
Although I like the conciseness of using this struct to avoid writing 5 tests, it might make debugging slightly more difficult if a test fails in the future, as the test framework will point to "check_precharge_state_transitions_and_air_plus_status", but won't indicate which step failed without manual debugging. I can live with it as is, but an FYI for future tests

Reply:
Yeah this is a good point, seems like a bit of a limitation of gtest that there's no easy way to do stuff like this
I don't love repeating the test 5 times since then we get massive testing files, and if we have to change the test down the line then we have to modify it in 5 places
But you're right there's probably a better way, like breaking it into a few test cases (all the reasons precharge will/won't start, slow precharge, fast precharge) and then do some smaller parameterizing afterwards 
But I'll leave it as-is for this PR

Reply:
Would also be nice maybe to centralize the main testing fixture so it's really easy to add seperate test files, and then we can get add lots of cases without getting mega files